### DISCUSSION TITLE
ChangeLog

### DISCUSSION THREAD

**Discussion by GitMensch** [WebFreak001/code-debug#355] [General]
Github doesn't add reference links in the CHANGELOG.md and a client using the extension can't.

Should we change the file to use explicit references (at the end of the block) for the issues and possibly pull requests (not sure about the later) as we do with the version comparisons?

I've added the comparison links using the sha sums where possible (and useful), the versions that don't have a useful git reference were kept out.
Would someone please take the time to use the sub-categories for "type of changes" (from https://keepachangelog.com/) for the old versions to fully "convert" the file to the new format?

**Comment by brownts** [comment]
Yeah, I noticed that too.  It seems like the marketplace website does add those links, but GitHub doesn't.  I don't see any reason that they can't be added.  That would be consistent with others that use this format (e.g., [GitLens](https://github.com/gitkraken/vscode-gitlens/blob/main/CHANGELOG.md)).

Additionally, if we are moving in that direction, the changelog probably should be updated structurally to follow the recommended formatting.  That means shifting the existing headers down a level and adding the level 1 heading at the top with the boilerplate paragraph, as documented on their website.  Also, dates should be added to each release section as well.

If you want, you can also create the tags for the actual release numbers instead of using the SHA.  I did that for the v0.25.1 tag, but didn't do it for the rest of them.  That might make it easier for the comparison links.
